go-controller/pkg/cni/cni.go

```diff
@@ -174,9 +174,31 @@ func (pr *PodRequest) cmdAddWithGetCNIResultFunc(kubeAuth *KubeAPIAuth, clientse
 			if err != nil {
 				return nil, err
 			}
-			if _, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo); err != nil {
+			primaryUDNResult, err := getCNIResultFn(primaryUDNPodRequest, clientset, primaryUDNPodInfo)
+			if err != nil {
 				return nil, err
 			}
+
+			defaultClusterNetworkResult := current.Result{
+				CNIVersion: response.Result.CNIVersion,
+				Interfaces: response.Result.Interfaces,
+				IPs:        response.Result.IPs,
+				Routes:     response.Result.Routes,
+				DNS:        response.Result.DNS,
+			}
+			originalNumberOfIfaces := len(primaryUDNResult.Interfaces)
+			response.Result.Interfaces = append(primaryUDNResult.Interfaces, response.Result.Interfaces...)
+			response.Result.IPs = append([]*current.IPConfig{}, primaryUDNResult.IPs...)
+			for i := range defaultClusterNetworkResult.IPs {
+				ifaceIPConfig := defaultClusterNetworkResult.IPs[i].Copy()
+				if ifaceIPConfig.Interface != nil {
+					ifaceIPConfig.Interface = current.Int(*ifaceIPConfig.Interface + originalNumberOfIfaces)
+				}
+				response.Result.IPs = append(response.Result.IPs, ifaceIPConfig)
+
+			}
+
+			response.Result.Routes = append(primaryUDNResult.Routes, response.Result.Routes...)
```

Comment:
It's a bit confusing. I would probably just do something more like:
```
response.Result.Routes = append(primaryUDNResult.Routes, response.Result.Routes...)
response.Result.Interfaces = append(primaryUDNResult.Interfaces, response.Result.Interfaces...)
response.Result.IPs = append(primaryUDNResult.IPs, response.Result.IPs... )

# offset the index of the default network IPs to correctly point to the default network interfaces
numOfPrimaryIfaces := len(primaryUDNResult.Interfaces)
for i := len(primaryUDNResult.IPs); i < len(response.Result.IPs); i++ {
    if response.Result.IPs[i].Interface != nil {
        response.Result.IPs[i].Interface = current.Int(*ifaceIPConfig.Interface + numOfPrimaryIfaces)
    }
}
```

Reply:
yeah I agree, your's much simpler. 
DONE